monitoring/check_collectd_mlab_test.py

```diff
@@ -0,0 +1,379 @@
+        if self._failure_mode == 'send_bad_reply':
+          client.sendall('not-a-number junk\n')
+        elif self._failure_mode == 'send_empty_reply':
+          client.sendall('\n')
+        else:
+          client.sendall('1 default reply\n')
+
+        # Send "actual values" response.
+        if self._failure_mode == 'close_after_reply':
+          # Skip the response, so client read will fail.
+          return
+        try:
+          client.sendall('default=1.0\n')
+        except socket.error:
+          # Incase of sigpipe due to client closing connection first.
+          pass
+    # pylint: enable=E1101
+
+
+class MlabNagiosCollectdTests(unittest.TestCase):
+
+  def setUp(self):
+    self._testdata_dir = os.path.join(
+        os.path.dirname(check_collectd_mlab.__file__), 'testdata')
+    self.fake_socket = os.path.join(self._testdata_dir, 'fake_socket')
+    self.fake_collectd = FakeCollectd(self.fake_socket)
+    self.fake_collectd.start()
+
+  def tearDown(self):
+    self.assertTrue(self.fake_collectd.shutdown())
+
+  def testunit_sock_sendcmd_RETURNS_successfully(self):
+    sock = check_collectd_mlab.sock_connect(self.fake_socket)
+
+    returned_value = check_collectd_mlab.sock_sendcmd(sock, 'GETVAL "whatever"')
+
+    self.assertEqual(returned_value, 1)
+
+  def testunit_sock_sendcmd_WHEN_receive_bad_reply_RETURNS_zero(self):
+    self.fake_collectd.set_response_behavior('send_bad_reply')
+    sock = check_collectd_mlab.sock_connect(self.fake_socket)
+
+    returned_value = check_collectd_mlab.sock_sendcmd(sock, 'GETVAL "whatever"')
+
+    self.assertEqual(returned_value, 0)
+
+  def testunit_assert_collectd_responds_WHEN_sock_sendcmd_fails(self):
+    self.fake_collectd.set_response_behavior('send_empty_reply')
+    check_collectd_mlab.COLLECTD_PID = (
+        os.path.join(self._testdata_dir, 'fake_pid'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = self.fake_socket
+
+    self.assertRaises(
+        check_collectd_mlab.SocketSendCommandCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+
+  def testunit_assert_collectd_responds_WHEN_sock_readline_fails(self):
+    self.fake_collectd.set_response_behavior('close_after_reply')
+    check_collectd_mlab.COLLECTD_PID = (
+        os.path.join(self._testdata_dir, 'fake_pid'))
+    check_collectd_mlab.COLLECTD_UNIXSOCK = self.fake_socket
+
+    self.assertRaises(
+        check_collectd_mlab.SocketReadlineCriticalError,
+        check_collectd_mlab.assert_collectd_responds)
+
+
+class MlabNagiosTests(unittest.TestCase):
+
+  def setUp(self):
+    self._testdata_dir = os.path.join(
+        os.path.dirname(check_collectd_mlab.__file__), 'testdata')
+
+  def testunit_sock_connect_WHEN_no_socket_RAISES_CriticalError(self):
+    self.assertRaises(
+        check_collectd_mlab.SocketConnectionCriticalError,
+        check_collectd_mlab.sock_connect, 'no_socket_name')
+
+  def testunit_sock_readline_WHEN_socket_error_RETURNS_empty_string(self):
```

Comment:
Test name doesn't seem to match behavior here because it's raising an error instead of returning empty string.